src/darwin/Framework/CHIP/MTRBaseDevice.mm

```diff
@@ -2290,20 +2290,32 @@ + (MTRAttributeRequestPath *)requestPathWithEndpointID:(NSNumber * _Nullable)end
 
 - (BOOL)isEqualToAttributeRequestPath:(MTRAttributeRequestPath *)path
 {
-    if (!path)
+    if (!path) {
         return NO;
+    }
+
+    if (path == self) {
+        return YES;
+    }
+
+    // Note: Nil is a wildcard, and empty endpoint/cluster/attribute values imply nil as well.
+    // Pass iff we have objects that are pointer equivalent (eg: nil == nil || object_a == object_a), or they have values and the same logical values
+    if ((_endpoint == path.endpoint || (path.endpoint && [_endpoint isEqualToNumber:path.endpoint]))
+        && (_cluster == path.cluster || (path.cluster && [_cluster isEqualToNumber:path.cluster]))
+        && (_attribute == path.attribute || (path.attribute && [_attribute isEqualToNumber:path.attribute]))) {
+        return YES;
+    }
 
-    return (path.endpoint && [_endpoint isEqualToNumber:path.endpoint])
-        && (path.cluster && [_cluster isEqualToNumber:path.cluster])
-        && (path.attribute && [_attribute isEqualToNumber:path.attribute]);
+    return [super isEqual:path];
```

Comment:
Shouldn't this just return NO?  The other thing is an instance of MTRAttributeRequestPath but the members don't match us.  Who cares what `super` has to say?

Reply:
That is correct, as we discussed, this is a redundant thanks to the self == path above.